tests/BenchmarkDotNet.Tests/Mocks/MockHostEnvironmentInfo.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Horology;
+
+namespace BenchmarkDotNet.Tests.Mocks
+{
+    public class MockHostEnvironmentInfo : HostEnvironmentInfo
+    {
+        public MockHostEnvironmentInfo()
```

Comment:
I suggest to rewrite this code in the following way:
```cs
public static readonly HostEnvironmentInfo Default = new MockHostEnvironmentInfo
{
    BenchmarkDotNetCaption = "BenchmarkDotNet",
    BenchmarkDotNetVersion = "0.x.y-mock",
    OsVersion = "MockOS",
    ProcessorName = new Lazy<string>(() => "MockProcessor CPU 42.0GHz"),
    ProcessorCount = 42,
    RuntimeVersion = "Clr 4.0.x.y",
    Architecture = "64bit",
    HasAttachedDebugger = false,
    HasRyuJit = true,
    Configuration = "RELEASE",
    JitModules = "clrjit-v4.6.x.y",
    DotNetCliVersion = new Lazy<string>(() => "1.0.0-mock-42"),
    ChronometerFrequency = new Frequency(42),
    ChronometerResolution = TimeInterval.FromNanoseconds(42),
    HardwareTimerKind = HardwareTimerKind.Tsc
};

private MockHostEnvironmentInfo()
{
}
```
1. It allows using a single instance of this class everywhere.
2. It allows introducing other instances of MockSummary.
3. The `Default` type is `HostEnvironmentInfo` instead of `MockHostEnvironmentInfo` which is better because we use `HostEnvironmentInfo` in the core method signatures.
4. It should be easy to see that we are working with the mock data. For example, if somebody accidentally prints the real environment data in an exporter, it shouldn't be equal to the mock data.
I made the above changes and found a bug. Here is my `ApprovalTest.Exporter.Md_StackOverflow.received.txt`:
```

    BenchmarkDotNet=v0.x.y-mock, OS=MockOS
    Processor=MockProcessor CPU 42.0GHz, ProcessorCount=42
    Frequency=42 Hz, Resolution=42.0000 ns, Timer=TSC
      [Host] : Clr 4.0.x.y, 64bit RyuJIT-v4.6.1586.0

    Job=LongRun  LaunchCount=3  TargetCount=100  
    WarmupCount=15  Allocated=N/A  

     Method |      Mean |    StdDev |       P67 |
    ------- |---------- |---------- |---------- |
        Foo | 1.0000 ns | 0.0000 ns | 1.0000 ns |
        Bar | 1.0000 ns | 0.0000 ns | 1.0000 ns |
```
As you can see, RyuJIT version is `4.6.1586.0` instead of `4.6.x.y` which is declared in `JitModules`. Thus, the test will fail for some version of the .NET Framework (with another RyuJIT version).